proxy/src/coco.rs

```diff
@@ -1,649 +1,358 @@
-/// extracted from a signature.
-pub struct Person {
-    /// Name part of the commit signature.
-    pub name: String,
-    /// Email part of the commit signature.
-    pub email: String,
-    /// Reference (url/uri) to a persons avatar image.
-    pub avatar: String,
-}
-
-/// Representation of a code commit.
-pub struct Commit {
-    /// Identifier of the commit in the form of a sha1 hash. Often referred to as oid or object
-    /// id.
-    pub sha1: git2::Oid,
-    /// The author of the commit.
-    pub author: Person,
-    /// The summary of the commit message body.
-    pub summary: String,
-    /// The entire commit message body.
-    pub message: String,
-    /// The committer of the commit.
-    pub committer: Person,
-    /// The recorded time of the committer signature. This is a convenience alias until we
-    /// expose the actual author and commiter signatures.
-    pub committer_time: git2::Time,
-}
-
-impl Commit {
-    /// Returns the commit description text. This is the text after the one-line summary.
-    #[must_use]
-    pub fn description(&self) -> &str {
-        self.message
-            .strip_prefix(&self.summary)
-            .unwrap_or(&self.message)
-            .trim()
+    async fn resolve_revision(
+        &self,
+        uri: &RadUrn,
+        _revision: u64,
+    ) -> Result<project::Project<entity::Draft>, entity::Error> {
+        let projects = self.projects.lock().expect("failed to acquire lock");
+        Ok(projects.get(uri).expect("project was missing").clone())
     }
 }
 
-impl From<&surf::git::Commit> for Commit {
-    fn from(commit: &surf::git::Commit) -> Self {
-        let avatar = |input: &String| {
-            let mut s = DefaultHasher::new();
-            input.hash(&mut s);
-
-            format!(
-                "https://avatars.dicebear.com/v2/jdenticon/{}.svg",
-                s.finish().to_string()
-            )
-        };
-
-        Self {
-            sha1: commit.id,
-            author: Person {
-                name: commit.author.name.clone(),
-                email: commit.author.email.clone(),
-                avatar: avatar(&commit.author.email),
-            },
-            summary: commit.summary.clone(),
-            message: commit.message.clone(),
-            committer: Person {
-                name: commit.committer.name.clone(),
-                email: commit.committer.email.clone(),
-                avatar: avatar(&commit.committer.email),
-            },
-            committer_time: commit.author.time,
-        }
+impl Peer {
+    /// We create a default `Peer` using the `tmp_dir_path` we provide.
+    pub async fn new<I>(
+        config: net::peer::PeerConfig<discovery::Static<I, SocketAddr>>,
+    ) -> Result<Self, error::Error>
+    where
+        I: Iterator<Item = (peer::PeerId, SocketAddr)> + Send + 'static,
+    {
+        // Initialise the storage
+        let _ = storage::Storage::init(&config.paths, config.key.clone())?;
```

Comment:
What's does it return?

Reply:
A `Storage`, but we're doing it for the side-effect of initialising the mono-repo. I think it should really be done via something like https://github.com/radicle-dev/radicle-link/issues/160